$NetBSD: patch-ab,v 1.13 2015/02/18 10:34:57 wiz Exp $

Add a missing comma that resulted in a smaller freqctl[] and setctl[] (and so
with a potential off-by-one error).
Explicitly use TECH_MAX in sizes of freqlist and related arrays.

--- estd.c.orig	2015-02-15 21:12:10.000000000 +0000
+++ estd.c
@@ -121,7 +121,7 @@ int             ndomains;
  static const size_t cp_time_max_size = sizeof(cp_time[0]) * MAX_CPUS;
 #endif
 
-static char	*techdesc[8] = {"Unknown",
+static char	*techdesc[TECH_MAX + 1] = {"Unknown",
 				"Enhanced SpeedStep",
 				"PowerNow",
 				"ACPI P-States",
@@ -130,20 +130,20 @@ static char	*techdesc[8] = {"Unknown",
 				"Rockchip",
 				"Generic"
 				};
-static char	*freqctl[8] = {	"",	
+static char	*freqctl[TECH_MAX + 1] = {	"",	
 				"machdep.est.frequency.available",
 				"machdep.powernow.frequency.available",
 				"hw.acpi.cpu.px_dom0.available",
-				"machdep.intrepid.frequency.available"
+				"machdep.intrepid.frequency.available",
 				"machdep.loongson.frequency.available",
 				"machdep.cpu.frequency.available",
 				"machdep.frequency.available"
 				};
-static char	*setctl[8] = {	"",
+static char	*setctl[TECH_MAX + 1] = {	"",
 				"machdep.est.frequency.target",
 				"machdep.powernow.frequency.target",
 				"hw.acpi.cpu.px_dom0.select",
-				"machdep.intrepid.frequency.target"
+				"machdep.intrepid.frequency.target",
 				"machdep.loongson.frequency.target",
 				"machdep.cpu.frequency.target",
 				"machdep.frequency.current"
